<a href="https://colab.research.google.com/github/EricAndradeNJIT/data-analytics-folder/blob/main/Height_Prediction_Parents_Height.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import files
heights = files.upload()

Saving heights.csv to heights.csv


In [2]:
#import packages

import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score, mean_squared_error as MSE

In [4]:
#read and rename csv

heights = pd.read_csv("heights.csv")
heights.head()

Family  Father  Mother Gender  Height  Kids  MidParent
0       1    78.5    67.0      M    73.2     4      75.43
1       1    78.5    67.0      F    69.2     4      75.43
2       1    78.5    67.0      F    69.0     4      75.43
3       1    78.5    67.0      F    69.0     4      75.43
4       2    75.5    66.5      M    73.5     4      73.66

In [6]:
#split data into features of interest (mother and father)

X = heights[['Mother','Father','Gender']]
y = heights[['Height']]

print(X)
print(y)

     Mother  Father Gender
0      67.0    78.5      M
1      67.0    78.5      F
2      67.0    78.5      F
3      67.0    78.5      F
4      66.5    75.5      M
..      ...     ...    ...
893    65.0    68.5      M
894    65.0    68.5      M
895    65.0    68.5      F
896    65.0    68.5      F
897    65.0    68.5      F

[898 rows x 3 columns]
     Height
0      73.2
1      69.2
2      69.0
3      69.0
4      73.5
..      ...
893    68.5
894    67.7
895    64.0
896    63.5
897    63.0

[898 rows x 1 columns]


In [7]:
#One hot gender column

one_hot = pd.get_dummies(data=X, columns=['Gender'])

X = pd.DataFrame(one_hot)

print (X)

     Mother  Father  Gender_F  Gender_M
0      67.0    78.5         0         1
1      67.0    78.5         1         0
2      67.0    78.5         1         0
3      67.0    78.5         1         0
4      66.5    75.5         0         1
..      ...     ...       ...       ...
893    65.0    68.5         0         1
894    65.0    68.5         0         1
895    65.0    68.5         1         0
896    65.0    68.5         1         0
897    65.0    68.5         1         0

[898 rows x 4 columns]


In [10]:
#split data into training/validating dataset and test dataset

X_train_val, X_test, y_train_val, y_test = train_test_split(X, y,test_size=0.25, random_state=42)

print(X_train_val)

     Mother  Father  Gender_F  Gender_M
377    62.0    70.5         1         0
357    63.0    70.5         1         0
723    64.0    67.0         0         1
306    64.7    70.0         1         0
464    66.0    69.0         1         0
..      ...     ...       ...       ...
106    63.5    72.5         1         0
270    65.0    70.0         0         1
860    63.0    65.5         0         1
435    66.0    69.0         1         0
102    63.0    72.0         1         0

[673 rows x 4 columns]


In [13]:
#split training/validating dataset into training and validation sets

X_train, X_val, y_train, y_val = train_test_split(X_train_val, y_train_val, test_size=0.333, random_state=42)

print(X_val)

     Mother  Father  Gender_F  Gender_M
153    65.5    71.0         0         1
768    66.0    66.0         1         0
559    66.5    68.5         0         1
19     58.5    75.0         1         0
681    59.0    68.0         0         1
..      ...     ...       ...       ...
542    60.0    69.0         0         1
457    66.0    69.5         1         0
189    63.5    71.0         0         1
269    65.0    70.0         0         1
663    61.0    68.0         1         0

[225 rows x 4 columns]


In [14]:
#fit linear regression

reg = LinearRegression()

model = reg.fit(X_train, y_train)

In [15]:
#calculate predicted y values 

y_pred = model.predict(X_train)

y_pred_val = model.predict(X_val)

In [16]:
#interpret results by using training data

RMSE_train = MSE(y_train, y_pred, squared = False)

r2_train = r2_score(y_train, y_pred)

print(RMSE_train, r2_train)

2.1761132728276626 0.6390857198371704


The average difference between an observed child's height and the height predicted is 2.2 inches

About 64% of the variability in the height of adult children can be explained by the height of the parents, their gender and the family of origin.

In [17]:
#interprest results by using validation data

RMSE_val = MSE(y_val, y_pred_val, squared = False)
print(RMSE_val)

r2_val = r2_score(y_val, y_pred_val)
print(r2_val)

2.0693705905203355
0.6683120474277819


The average difference between an observed child's height and the height predicted is 2.1 inches

About 67% of the variability in the height of adult children can be explained by the height of the parents, their gender and the family of origin.